data-service/src/test/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDBTest.java

```diff
@@ -198,4 +255,197 @@ public void geEnvironment() throws StorageException, EnvironmentNotFoundExceptio
+    assertEquals(
+        expectedDynamodbScanExpression.getIndexName(),
+        dynamoDBScanExpressionCaptor.getValue().getIndexName());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void listEnvironmentIdsByClusterNullCluster() throws StorageException {
+    environmentRepositoryDDB.listEnvironmentIdsByCluster(null);
+  }
+
+  @Test(expected = StorageException.class)
+  public void listEnvironmentIdsByClusterStorageException() throws StorageException {
+    when(dynamoDBMapper.query(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBQueryExpression.class)))
+        .thenThrow(new AmazonServiceException(""));
+    environmentRepositoryDDB.listEnvironmentIdsByCluster(CLUSTER);
+  }
+
+  @Test
+  public void listEnvironmentIdsByCluster() throws StorageException {
+    List<EnvironmentTargetVersionDDBRecord> environmentTargetVersionDDBRecords = new ArrayList<>();
+    environmentTargetVersionDDBRecords.add(environmentTargetVersionDDBRecord);
+    final EnvironmentTargetVersionDDBRecord secondRecord =
+        EnvironmentTargetVersionDDBRecord.builder()
+            .environmentId(SECOND_ENVIRONMENT_ID)
+            .environmentVersion(UUID.randomUUID().toString())
+            .cluster(environmentTargetVersionDDBRecord.getCluster())
+            .build();
+    environmentTargetVersionDDBRecords.add(secondRecord);
+    when(environmentTargetVersionDDBRecordPaginatedQueryList.stream())
+        .thenReturn(environmentTargetVersionDDBRecords.stream());
+
+    when(dynamoDBMapper.query(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBQueryExpression.class)))
+        .thenReturn(environmentTargetVersionDDBRecordPaginatedQueryList);
+
+    final List<String> environmentsByCluster =
+        environmentRepositoryDDB.listEnvironmentIdsByCluster(CLUSTER);
+
+    assertEquals(environmentTargetVersionDDBRecords.size(), environmentsByCluster.size());
+    for (EnvironmentTargetVersionDDBRecord record : environmentTargetVersionDDBRecords) {
+      if (!environmentsByCluster.contains(record.getEnvironmentId())) {
+        fail(String.format("Does not contain %s", record.getEnvironmentId()));
+      }
+    }
+
+    verify(dynamoDBMapper)
+        .query(
+            eq(EnvironmentTargetVersionDDBRecord.class), dynamoDBQueryExpressionCaptor.capture());
+    final DynamoDBQueryExpression<EnvironmentTargetVersionDDBRecord>
+        expectedDynamodbQueryExpression =
+            new DynamoDBQueryExpression<EnvironmentTargetVersionDDBRecord>()
+                .withIndexName(EnvironmentTargetVersionDDBRecord.ENVIRONMENT_CLUSTER_GSI_NAME)
+                .withConsistentRead(false)
+                .withHashKeyValues(
+                    EnvironmentTargetVersionDDBRecord.withGSIHashKey(
+                        environmentTargetVersionDDBRecord.getCluster()));
+
+    assertEquals(
+        expectedDynamodbQueryExpression.getIndexName(),
+        dynamoDBQueryExpressionCaptor.getValue().getIndexName());
```

Comment:
If we're not comparing the two expression objects directly, why not use the values directly instead of constructing the intermediate expression?

Reply:
how do you use the values directly? if i just use the expected in verify it compares the objects and throws something to the effect of DynamoDBScanExpression@2c88b9fc != DynamoDBScanExpression@3918c187

Reply:
Yeah, what I mean is, why even construct `expectedDynamoDbQueryExpression` in the previous line if you're not going to use it? Rather use the value directly.

Reply:
Sure that would work too but this works as well and I don't think it warrants another change.